code/src/test/kotlin/com/expediagroup/sdk/core/http/RequestTest.kt

```diff
@@ -0,0 +1,280 @@
+package com.expediagroup.sdk.core.http
+
+import java.net.MalformedURLException
+import java.net.URL
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNull
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class RequestTest {
+
+    @Test
+    fun `should build request instance with all properties`() {
+        // Given
+        val method = Method.POST
+        val url = URL("https://example.com")
+        val headers = Headers.Builder().add("Authorization", "Bearer token").build()
+        val body = RequestBody.create("Sample body".byteInputStream())
+
+        // When
+        val request = Request
+            .builder()
+            .method(method)
+            .url(url)
+            .headers(headers)
+            .body(body)
+            .build()
+
+        // Expect
+        assertEquals(method, request.method)
+        assertEquals(url, request.url)
+        assertEquals("Bearer token", request.headers.get("Authorization"))
+        assertEquals(body, request.body)
+    }
+
+    @Test
+    fun `should build a new request based on previous instance`() {
+        // Given
+        val originalRequest = Request
+            .builder()
+            .method(Method.GET)
+            .url("https://example.com")
+            .addHeader("Content-Type", "application/json")
+            .build()
+
+        // When
+        val newRequest = originalRequest
+            .newBuilder()
+            .addHeader("Accept", "text/plain")
+            .build()
+
+        // Expect
+        assertEquals(Method.GET, newRequest.method)
+        assertEquals(originalRequest.url, newRequest.url)
+
+        assertEquals("application/json", originalRequest.headers.get("Content-Type"))
+        assertNull(originalRequest.headers.get("Accept"))
+
+        assertEquals("application/json", newRequest.headers.get("Content-Type"))
+        assertEquals("text/plain", newRequest.headers.get("Accept"))
+    }
+
+    @Test
+    fun `should build a request with valid method and url string`() {
+        // Given
+        val urlString = "https://example.com"
+
+        // When
+        val request = Request
+            .builder()
+            .method(Method.GET)
+            .url(urlString)
+            .build()
+
+        // Expect
+        assertEquals(URL(urlString), request.url)
+    }
+
+    @Test
+    fun `should add single header`() {
+        // Given
+        val method = Method.GET
+        val url = "https://example.com"
+        val headerName = "content-type"
+        val headerValue = "application/json"
+
+        // When
+        val request = Request
+            .builder()
+            .method(method)
+            .url(url)
+            .addHeader(headerName, headerValue)
+            .build()
+
+        // When
```

Comment:
Expect*

Please also update it for the rest of the cases in the file.

Reply:
Done